server/chat-plugins/roomsettings.js

```diff
@@ -328,6 +329,8 @@ exports.commands = {
 			if (!this.can('editroom', null, room)) return;
 		} else if (room.battle) {
 			if (!this.can('editprivacy', null, room)) return;
+			const prefix = room.battle.forcedPublic();
+			if (prefix && !user.can('lock')) return this.errorReply(`This battle is required to be public due to a player having a name prefixed by '${prefix}'.`);
```

Comment:
`lock` seems like a weird permission for this, personally

Reply:
Basically just wanted gstaff to be able to handle things. I guess global `editprivacy` makes the most sense, but its `&`+. Switched to that, don't know if I'll have people in staff getting angry with me.